src/backend/api/meals.js

```diff
@@ -2,14 +2,97 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+//-------------------------------------------
+
+//http://localhost:5000/api/meals/title
+router.get("/title", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
+    const titles = await knex("meal").select("title");
     response.json(titles);
   } catch (error) {
     throw error;
   }
 });
 
+//-------------------------------------------
+
+// GET /api/meals  - Returns all meals
+router.get("/", async (req, res) => {
+  try {
+    const allMeals = await knex.select("*").from("meal");
+    if (allMeals) {
+      res.status(200).json(allMeals);
+    } else {
+      res.status(404).send("The meals were not found");
```

Comment:
So here again - we need to think a bit about whether the fact that no meals were found was really an error (there should always be some meals) or is just a state that our application might be in (for example no one added any meals yet). If it is the latter that is correct, then it is better if an endpoint returns a JSON with an empty array inside, indicating that we simply do not have any meals yet, instead of an error. That way the response we send to the client has a consistent model.